omod/src/main/java/org/openmrs/web/user/CurrentUsers.java

```diff
@@ -83,13 +91,14 @@ public static void addUser(HttpSession httpSession, User user) {
 	 * @param httpSession
 	 */
 	public static void removeUser(HttpSession httpSession) {
-		String sessionId = httpSession.getId();
-		Map<String, String> currentUsers = getCurrentUsers(httpSession);
-		if (log.isDebugEnabled()) {
-			log.debug("Removing user from the current users. session: " + sessionId + " user: "
-			        + currentUsers.get(sessionId));
+		User currentUser = Context.getAuthenticatedUser();
+		if (currentUser != null) {
+			String userName = getValidUserName(currentUser);
+			Map<String, CopyOnWriteArrayList<HttpSession>> currentUsers = getCurrentUsers(httpSession);
+			invalidateAllUserSessions(userName, currentUsers);
```

Comment:
Do we actually want to invalidate *all* of the user's sessions when the log out of a single session? I would suggest instead that we simply remove the user from the currentUsers map when the number of sessions remaining is 0.

Reply:
I aimed at invalidating all user sessions on password change.  I created this method so that I use it in the adminui-module, and invalidate all user sessions after a password change.  

The on logout, the request is always invalidated which calls the listener and the list of sessions is updated

Reply:
So, I totally buy that we need a method that invalidates all user sessions and that makes a lot of sense. What doesn't make sense to me is killing all of a user's sessions every time one of their sessions expires (this is different from "what happens when I change my password" where it makes sense to kill all of the users sessions). That is to say that the password reset code probably needs to have a special case to invalidate all of the sessions.

Reply:
In simple terms each time the method `removeUser(httpSession)` is used, it will invalidate all sessions for the requested user.  
Each time a request is invalidated like https://github.com/openmrs/openmrs-module-legacyui/blob/master/omod/src/main/java/org/openmrs/web/servlet/LogoutServlet.java#L46 the Listener is executed and that session is removed from the sessions list of a specified user to prevent references to null objects in the list.